Reorganized commands under a main command, added list command

bippity opened the pull request:
Cleaned up commands to include help text/description and under a main command (/warpplate, /wp).
* Reformatted commands into sub-commands
* Added command to list all warpplates in the current world (/wp list)

Followed the format made in TShock for the group/region/warp commands. Built with TShock pre-release 15.

bippity committed:
Added list warpplates command

bippity committed:
Reformatted into sub-commands

popstarfreas commented on Commands.cs:
```diff
@@ -19,6 +21,122 @@ public Commands(WarpplateManager manager, Player[] players)
+
+                case "dim":
+                case "dimension":
+                    SetWarpplateDimension(args);
```
The permission has been changed from `warpplate.setdimensional` -> `warpplate.set` which seems unintentional given the permission check in the help section.

bippity replied:
Good catch, missed it while testing the plugin at 5am :)
Resolved in 49bfcc5d3d5519cb2a7056a8b3afc44faa2ee300

popstarfreas commented on WarpplateManager.cs:
```diff
@@ -74,9 +74,9 @@ public string InAreaWarpplateName(int x, int y)
+        public List<String> ListAllWarpplates()
         {
-            var WarpplatesTemp = new List<Warpplate>();
+            /*var WarpplatesTemp = new List<Warpplate>();
```
You should probably delete instead of commenting out code that has been replaced.

bippity replied:
Resolved in c68d4b0fd1334e48c192e65f101f8c90565f911c

popstarfreas reviewed (commented):
I don't have time to check through everything but I commented on what I found

bippity committed:
Fixed missing permission check for dimension

bippity committed:
Removed commented out code

bippity committed:
Added separate command for player use

Added /wpa specifically for player use. Kept the toggle sub-command for admin use.

popstarfreas commented on Commands.cs:
```diff
@@ -19,6 +21,127 @@ public Commands(WarpplateManager manager, Player[] players)
+                case "dimension":
+                    if (!args.Player.HasPermission("warpplate.setdimensional"))
+                    {
+                        args.Player.SendErrorMessage("You do not have permission to set dimensions.");
```
Can this be "You do not have permission to set a dimensional warpplate." ?

bippity replied:
Resolved

bippity committed:
Revised error message

Revised dimension error message for more clarity

bippity committed:
TShock 4.5.0.1 Update

popstarfreas commented:
I don't have enough time to look through this for most likely the next 20 days, but can you make sure that this PR contains one unit of change; so don't update any of the refs in this PR.

bippity committed:
Revert "TShock 4.5.0.1 Update"

This reverts commit b172fc1bc2b048fdc305b64caf89f45432ffd260.
